src/main/java/org/magicdgs/hdf5j/fileformat/FileAddress.java

```diff
@@ -0,0 +1,137 @@
+package org.magicdgs.hdf5j.fileformat;
+
+import org.magicdgs.hdf5j.utils.exceptions.FileAddressException;
+
+import com.google.common.base.Preconditions;
+
+import java.io.IOException;
+import java.nio.channels.SeekableByteChannel;
+
+/**
+ * Representation of a file address (position to seek).
+ *
+ * @author Daniel Gomez-Sanchez (magicDGS)
+ */
+public interface FileAddress {
+
+    /** Value for {@link #getFilePointer()} for the undefined address. */
+    public static final long UNDEFINED_FILE_POINTER = -1;
+
+    /**
+     * Gets the file pointer for this address.
+     *
+     * <p>This is the long representation for the file address to be used for seek a file. For
+     * example, using {@link SeekableByteChannel#position(long)}.
+     *
+     * @return file position represented by this address (strictly positive);
+     * {@link #UNDEFINED_FILE_POINTER} if undefined.
+     */
+    public long getFilePointer();
+
+    /**
+     * Gets the byte array representation for this address, in big-endian order.
+     *
+     * <p>The length of the array should be the same as the value returned by
+     * {@link #getNumberOfBytes()}.
+     *
+     * <p><b>WARNING</b>: implementations of this method are not forced to return an unmodifiable
+     * byte array. Do not modify the returned array unless sub-classes specify that it is safe.
+     *
+     * @return byte array in big-endian order to encode the address.
+     */
+    public byte[] asByteArray();
+
+
+    /**
+     * Gets the number of bytes used to encode this file address into a byte array with {@link
+     * #asByteArray()}.
+     *
+     * @return the number of bytes used to encode this file address position.
+     */
+    public int getNumberOfBytes();
+
+    /**
+     * Returns the hexadecimal representation of the address from the byte array.
+     *
+     * <p>Default implementation calls {@code hexDisplay(asByteArray())}.
+     *
+     * @return address hexadecimal representation.
+     */
+    public default String hexDisplay() {
```

Comment:
If the `asByteArray` signature changes, remove default behavior and point to the static method for a common implementation.

Reply:
Done.